### PR TITLE
Remove mina-rs dependency

### PR DESCRIPTION
**Pull request by jhult**
Fixes https://github.com/Granola-Team/mina-indexer/issues/342

I have not yet explored the overlap with `mina_blocks`. That will be handled in a second PR.

### CODE REVIEW

**Comment by trevorbernard** [comment]
@jhult I wonder if we can't turn off clippy for the mina-rs code

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,204 @@
...
+*.cab
+*.msi
+*.msix
+*.msm
+*.msp
+
+# Windows shortcuts
+*.lnk
+
+# End of https://www.toptal.com/developers/gitignore/api/rust,linux,macos,windows,intellij+iml,rust-analyzer,visualstudiocode
```
Comment (trevorbernard) [inline .gitignore:204]: This `.gitignore` doesn't understand our application on some of it wrong. Like removing the previous entries and ignoring things like `Cargo.lock`. I would revert back to the previous `.gitignore` file and add only what needs to be ignored. E.g. Intellij project files

**Review by trevorbernard — CHANGES_REQUESTED** [review]

**Comment by trevorbernard** [comment]
Also, I would also update documentation that refers to git submodules. Like in the readme for nix.

```
nix build .?submodules=1
nix build
```

And I would remove `git submodule update --init --recursive` from the shell hook

File: .gitignore (hunk 1 above)
Comment (jhult) [reply .gitignore:204]: I will revert this.

FWIW, I tend to use `.gitignore` files based on [toptal.com/developers/gitignore](https://www.toptal.com/developers/gitignore/). However, these aren't perfect and often need tweaks depending on the project.

**Review by trevorbernard — APPROVED** [review]

**Pull request merged**